nidm/nidm-results/scripts/create_nidmr_context.py

```diff
@@ -62,40 +64,47 @@ def main(owl=None):
     context['@context']['nlx'] = 'http://uri.neuinfo.org/nif/nifstd/'
     context['@context']['skos'] = 'http://www.w3.org/2004/02/skos/core#'
 
+
     #     sep = \
     #         '#################################################################'
 
     #     with open(self.file, 'r') as fp:
     #         owl_txt = fp.read()
     # For anything that has a label
+ 
     for s, o in sorted(owl.graph.subject_objects(SKOS['prefLabel'])):
-        json_key = str(o)
-        if '_' in json_key:
-            json_key = str(o).split('_')[1]
+        json_key = str(o)  
         context['@context'][json_key] = OrderedDict()
-
         if s in owl.ranges:
-            context['@context'][json_key]['@id'] = str(s)
-            context['@context'][json_key]['@type'] = next(iter(owl.ranges[s]))
+            if 'http://www.w3.org/2001/XMLSchema#int' in next(iter(owl.ranges[s])) or 'http://www.w3.org/2001/XMLSchema#double' in next(iter(owl.ranges[s])) or 'http://www.w3.org/2001/XMLSchema#integer' in next(iter(owl.ranges[s])) or  'http://www.w3.org/2001/XMLSchema#positiveInteger' in next(iter(owl.ranges[s])) : 
```

Comment:
To make the code easier to read, let's use a variable `ranges` to store what's in `next(iter(owl.ranges[s]))`.

```suggestion
            ranges = next(iter(owl.ranges[s]))
            if 'http://www.w3.org/2001/XMLSchema#int' in ranges or 'http://www.w3.org/2001/XMLSchema#double' in ranges or 'http://www.w3.org/2001/XMLSchema#integer' in ranges or  'http://www.w3.org/2001/XMLSchema#positiveInteger' in ranges: 
```